src/lib.rs

```diff
@@ -70,15 +70,30 @@ where
 fn sanitize_path(uri: &mut Uri) {
     let path = uri.path();
     let path_decoded = decode(path);
+
+    // Check if the path contains a trailing slash and that it is not the only
+    // character.
+    let trailing_slash = if let Some(char) = path_decoded.chars().last() {
+        char == '/' && path_decoded.len() > 1
+    } else {
+        false
+    };
```

Comment:
This could be greatly simplified with `is_some_and`, but it would bump the msrv of this crate to 1.70.

Reply:
nit: would something like this work?

```suggestion
    let trailing_slash = path_decoded.len() > 1 && path_decoded.chars().last().filter('/').is_some()
```